code/WorkInProgress/Electronics.dm

```diff
@@ -466,92 +462,253 @@
+	rkitFile.ruckData = ruck_controls
+	rkitFile.target = target
+	SPAWN_DBG(0.5 SECONDS)
+		var/datum/signal/newsignal = get_free_signal()
+		newsignal.source = src
+		newsignal.transmission_method = TRANSMISSION_RADIO
+		newsignal.data["command"] = "UPLOAD"
+		newsignal.data["address_1"] = target
+		newsignal.data["sender"] = src.net_id
+		newsignal.data_file = rkitFile
+		radio_connection.post_signal(src, newsignal)
+	known_rucks |= target
+	send_sync()
+
+//Run this if there's a file and return
+//This will either work, or you rejected a signal that had a file it didn't need
+/obj/machinery/rkit/proc/process_upload(datum/signal/signal)
+	var/target = signal.data["sender"]
+	var/command = signal.data["command"]
+	if(!target || (command != "add" && command != "UPLOAD") || (!istype(signal.data_file, /datum/computer/file/electronics_scan) && !istype(signal.data_file, /datum/computer/file/electronics_bundle)))
+		return
+	//If we get a database file, check that we just booted and that the file was made for us
+	//And also that we haven't already digested a database
+	var/datum/computer/file/electronics_bundle/rkitFile = signal.data_file
+	if (istype(rkitFile) && !data_initialized && !isnull(boot_time) && rkitFile.target == src.net_id)
+		var/datum/mechanic_controller/originalData = rkitFile.ruckData
+		data_initialized = 1
+		if(world.time - boot_time <= 3 SECONDS)
+			for (var/datum/electronics/scanned_item/O in originalData.scanned_items)
+				ruck_controls.scan_in(O.name, O.item_type, O.mats, O.locked) //Copy the database on digest so we never waste the effort
+			return
+		return
+	else if(istype(rkitFile))
+		return
+
+	//And then process blueprint files
+	//Scan them in if we haven't seen them before
+	//UPLOAD is the internal command and doesn't generate PDA messages
+	//add is sent by PDA scanners and does generate messages
+	var/datum/computer/file/electronics_scan/scanFile = signal.data_file
+
+	for(var/datum/electronics/scanned_item/O in ruck_controls.scanned_items)
+		if(scanFile.scannedPath == O.item_type)
+			if (command == "UPLOAD" || src.net_id != host_ruck) //Don't send a failure message if the it's an internal transfer("UPLOAD" command)
+				//And don't send a message if we're not the host
+				return //But we already had that blueprint, so we do leave
+
+			pda_message(target, "Notice: Item already in database.")
+
+			return
+	var/strippedName = scanFile.scannedName
+	ruck_controls.scan_in(strippedName, scanFile.scannedPath, scanFile.scannedMats)
+
+	if(src.net_id != host_ruck || command != "add") //Only the host sends PDA messages, and we don't send them for internal transfer
+		return
+
+	pda_message(target, "Notice: Item entered into database.")
 
 /obj/machinery/rkit/receive_signal(datum/signal/signal)
 	if(status & NOPOWER)
 		return
 
-	if(!signal || signal.encryption || !signal.data["sender"])
+	if(!signal || signal.encryption || !signal.data["sender"] || isnull(boot_time))
 		return
 
```

Comment:
Help command?

Reply:
Help command?

Reply:
https://github.com/goonstation/goonstation/blob/ef44529c2e43547d3de3c15abcd8cf1068602d2a/code/obj/machinery/computer/security/riot.dm#L68

Reply:
I'd this going to be documented on the wiki? If so I don't see why people have to necessarily rely on out of game knowledge. Your changes to multitool make it so you doing have to look up device frequencies for instance.

A lot of commercial and industrial grade equipment is intended to be repaired, serviced, troubleshot, and automated via debug ports. So feedback of some kind is often desired because most of those things aren't necessarily designed with malicious intent in mind and often thus can be a vector for attack. 🤷